llvm/lib/SYCLLowerIR/SanitizerKernelMetadata.cpp

```diff
@@ -38,6 +38,30 @@ PreservedAnalyses SanitizerKernelMetadataPass::run(Module &M,
   auto &DL = M.getDataLayout();
   auto &Ctx = M.getContext();
 
+  // Fix device global type, by wrapping a structure type
+  {
+    assert(KernelMetadata->getValueType()->isArrayTy());
+
+    auto Name = KernelMetadata->getName().str();
+    KernelMetadata->setName(Name + "_del");
+    auto *KernelMetadataOld = KernelMetadata;
```

Comment:
There is [`Value::takeName`](https://llvm.org/doxygen/classllvm_1_1Value.html#ae855357b6c5e6e7ed1869272708a3a84) which will simplify name transfer

Reply:
Done